packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -930,6 +930,14 @@ export const ComboBoxMixin = (subclass) =>
           this.opened || this.autoOpenDisabled
             ? this.$.dropdown.indexOfLabel(this.filter)
             : this._indexOfValue(this.value, this.filteredItems);
+
+        // see https://github.com/vaadin/web-components/issues/2615
+        if (this.value && this._focusedIndex >= 0) {
```

Comment:
These checks don't seem to limit the below invocation of `this._selectItemForValue(this.value);` to the case described in the issue linked above. Try adding a `console.log` after `_selectItemForValue`, and then run the combo-box tests. Seems that it's called quite a few times.

Also, removing `this.value` from this if-clause didn't seem to change anything / didn't break any tests, so there seems to be some issues in checking the desired state for this additional call to `_selectItemForValue`.

Reply:
You're right. I meant to have it run only when `selectedItem` is null. Changed it accordingly. Now it only gets called when setting the items or filteredItems while the comboBox has a value beforehand.